### DISCUSSION TITLE
import your nextjs-blog repository

### DISCUSSION THREAD

**Discussion by NZoumpas** [vercel/next.js#22010] [Help]
Hello
I have created a new repository according to the instructions, I followed from Learn about Nextjs.
I see in the repository there are only his files nextjs(pages, public, styles etc).The folders I create to complete the blog do not appear in the repository.And this is my question..I have made a mistake and the folders do not appear or this is the result of this cycle(Nextjs-blog)I am new to programming and any answer would be highly acceptable.I should also note that on my computer after running the application I get the desired result.Τhat this is the result nextjs-blog..and not the welcome page(Welcome to Nextjs!)
Kind regards,
Nikos

**Comment by mkrtchian** [comment]
Hello, what do you mean by "The folders I create [...] do not appear in the repository" ? Do you mean your local repository or a remote repository like on github ? If you mean a remote repository, then you need to add and commit your changes to your local repository before pushing them to the remote one:
```bash
git add .
git commit -m "your commit message"
```
It seems to me that your issue is related to git usage. You should maybe follow a tutorial on how to use it.

**Reply by NZoumpas** [reply]
Hello mkrtchian,
Thanks for your reply,
I create a new repository(new repository , Github)
from the terminal I give the following commands , git remote add origin https://... and also git push -u origin main.
That's enough to see the folders and files I see in visual studio code?
So far I see in the new repository only the original files and files from the same Νext.js.
that I need a seminar this is the only sure thing ...

**Reply by mkrtchian** [reply]
When you create a Next.js project with `npx create-next-app`, you get a folder that is also a git repository, with a first commit made for you, having all the files in the folder. It's your local repository. With the commands you said (git remote add ..., and git push ..), you can push the content of that local repository to a remote repository on github.

When you change or add files, the content of your git local repository does not change, you have to commit your new changes for that. Only then you will be able to push these changes and update the remote github repository with your local one.

Here is the flow with comments:
```bash
# 1. You change some files in your folders with your editor
# You can see the files that have changed:
git status # the list of the changed files
git diff # the detail of what you have changed
# 2. You add that files to the staging git environment 
git add my_changed_file_1 my_changed_file_2
# 3. You commit the staged environment
git commit -m "Explanation of the changes made"
# 4. You push your local branch to update the remote one
git push origin main
# And finally your github repository will have the changes made on my_changed_file_1 and my_changed_file_2
```

You can find a lot of articles and video tutorials on the web to learn git.
Here are two videos:
* [git crash course by Traversy Media](https://www.youtube.com/watch?v=SWYqp7iY_Tc)
* [git crash course by Mosh](https://www.youtube.com/watch?v=8JJ101D3knE)

**Reply by NZoumpas** [reply]
Hi mkrtchian,
Τhanks for the facility and for the tips for the new repository. I think I now have a much better understanding of this feature.
Αlso the videos in great detail
